java-frontend/src/test/files/se/NPE_reporting.java

```diff
@@ -58,4 +76,16 @@ private Object foo(boolean a) {
     }
     return new Object();
   }
+
+  class A {}
+
+  public void testMemberSelect(A a1, @CheckForNull A a2, @Nullable A a3) {
+    a1.hashCode(); // No issue
+    a2.hashCode(); // Noncompliant [[flows=a2]] {{NullPointerException might be thrown as 'a2' is nullable here}} flow@a2 {{a2 is dereferenced}}
+    a3.hashCode(); // Noncompliant [[flows=a3]] {{NullPointerException might be thrown as 'a3' is nullable here}} flow@a3 {{a3 is dereferenced}}
+
+    a1.a.hashCode(); // No issue
+    a1.b.hashCode(); // False negative
```

Comment:
This was copy&pasted from `java-frontend/src/test/files/se/NullDereferenceCheck.java` , because there were some issues with the flow in this. You're right, that this is not needed in this test. I am not sure of the original purpose.